crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -1551,23 +1558,52 @@ impl CumulocityConverter {
                 self.config.ops_dir.join(child_dir_name).into()
             }
             EntityType::Service => {
-                let target = &device.topic_id;
-                error!("Unsupported {c8y_operation_name} operation for a service: {target}");
-                return Ok(vec![]);
+                error!(
+                    %target,
+                    "{c8y_operation_name} operation for services are currently unsupported"
+                );
+                return Err(ConversionError::UnexpectedError(anyhow!(
+                    "{c8y_operation_name} operation for services are currently unsupported"
+                )));
             }
         };
+
         let ops_file = ops_dir.join(c8y_operation_name);
         create_directory_with_defaults(&*ops_dir)?;
-        create_file_with_defaults(ops_file, None)?;
+        create_file_with_defaults(&ops_file, None)?;
+
+        let c8y_prefix = self.config.bridge_config.c8y_prefix.clone();
+
+        let operation = c8y_api::smartrest::operations::get_operation(
+            ops_file.as_std_path(),
+            &self.config.bridge_config,
+        )?;
+        let operations = self
+            .operations_for_device_mut(target)
+            .expect("entity should've been checked before that it's not a service");
 
-        let need_cloud_update = self.update_operations(ops_dir.as_std_path())?;
+        let prev_operation = operations.remove_operation(&operation.name);
+        // even if the body of the operation is different, as long as it has the same name, supported operations message
+        // will be the same, so we don't need to resend
+        let need_cloud_update = prev_operation.is_none();
```

Comment:
Why not move this logic into the existing `update_operations` function, as that function already has the contract to return the boolean flag whether the cloud update is required or not?

Reply:
The function changed a bit in a313601, and I ended up with two branches, in the case of a child device I just call `update_operations` outright. Let me know if this addresses it, or if you had something more particular in mind.